filter.js

```diff
@@ -0,0 +1,7 @@
+(function() {
+    ele('filter').addEventListener('change', evt => {

```

Comment:
I'm feeling a bit sad that all the tests have to be reexecuted each time we select a new filter.
The UI becomes unresponsive for half a second when selecting "All".

Reply:
I'll see if I can get it to just execute once.

Sebastian

Reply:
I realize now that its more refactoring than I anticipated for this patch. The calculation and output logic are currently strongly entangled.

Let's split that out into a refactoring patch.

Sebastian

Reply:
Ok let's leave that for a follow-up PR